src/sinks/clickhouse/arrow_schema.rs

```diff
@@ -0,0 +1,767 @@
+
+use crate::http::{Auth, HttpClient};
+
+const DECIMAL32_PRECISION: u8 = 9;
+const DECIMAL64_PRECISION: u8 = 18;
+const DECIMAL128_PRECISION: u8 = 38;
+const DECIMAL256_PRECISION: u8 = 76;
+
+#[derive(Debug, Deserialize)]
+struct ColumnInfo {
+    name: String,
+    #[serde(rename = "type")]
+    column_type: String,
+}
+
+/// Fetches the schema for a ClickHouse table and converts it to an Arrow schema.
+pub async fn fetch_table_schema(
+    client: &HttpClient,
+    endpoint: &str,
+    database: &str,
+    table: &str,
+    auth: Option<&Auth>,
+) -> crate::Result<Arc<Schema>> {
+    // Query to get table schema
+    let query = format!(
+        "SELECT name, type \
+         FROM system.columns WHERE database = '{}' AND table = '{}' \
+         ORDER BY position FORMAT JSONEachRow",
+        database, table
+    );
+
+    let encoded_query =
+        percent_encoding::utf8_percent_encode(&query, percent_encoding::NON_ALPHANUMERIC)
+            .to_string();
+    let uri = format!("{}?query={}", endpoint, encoded_query);
+    let mut request = Request::get(&uri).body(Body::empty()).unwrap();
+
+    if let Some(auth) = auth {
+        auth.apply(&mut request);
+    }
+
+    let response = client.send(request).await?;
+
+    match response.status() {
+        StatusCode::OK => {
+            let body_bytes = http_body::Body::collect(response.into_body())
+                .await?
+                .to_bytes();
+            let body_str = String::from_utf8(body_bytes.into())
+                .map_err(|e| format!("Failed to parse response as UTF-8: {}", e))?;
+
+            parse_schema_from_response(&body_str)
+        }
+        status => Err(format!("Failed to fetch schema from ClickHouse: HTTP {}", status).into()),
+    }
+}
+
+/// Parses the JSON response from ClickHouse and builds an Arrow schema.
+fn parse_schema_from_response(response: &str) -> crate::Result<Arc<Schema>> {
+    let mut columns: Vec<ColumnInfo> = Vec::new();
+
+    for line in response.lines() {
+        if line.trim().is_empty() {
+            continue;
+        }
+
+        let column: ColumnInfo = serde_json::from_str(line)
+            .map_err(|e| format!("Failed to parse column info: {}", e))?;
+        columns.push(column);
+    }
+
+    if columns.is_empty() {
+        return Err("No columns found in table schema".into());
+    }
+
+    let mut fields = Vec::new();
+    for column in columns {
+        let (arrow_type, nullable) = clickhouse_type_to_arrow(&column.column_type)
+            .map_err(|e| format!("Failed to convert column '{}': {}", column.name, e))?;
+        fields.push(Field::new(&column.name, arrow_type, nullable));
+    }
+
+    Ok(Arc::new(Schema::new(fields)))
+}
+
+/// Unwraps ClickHouse type modifiers like Nullable() and LowCardinality().
+/// Returns a tuple of (base_type, is_nullable).
+/// For example: "Nullable(LowCardinality(String))" -> ("String", true)
+fn unwrap_type_modifiers(ch_type: &str) -> (&str, bool) {
```

Comment:
Is it possible to have `LowCardinality(Nullable(String))`? If so this logic fails